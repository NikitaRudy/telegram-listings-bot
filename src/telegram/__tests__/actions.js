const telegramActions = require('../actions');
const testUtils = require('../../../test-utils');

const botMock = {
  telegram: {
    sendPhoto: jest.fn(() => Promise.resolve()),
  },
};

const user = { chatId: '123124', username: 'johndoe' };

test('sendListingsToUser', async () => {
  await telegramActions.sendListingsToUser(botMock, user, testUtils.listings.mocks.listings);

  expect(botMock.telegram.sendPhoto).toHaveBeenCalled();
  expect(botMock.telegram.sendPhoto.mock.calls).toMatchSnapshot();
});
