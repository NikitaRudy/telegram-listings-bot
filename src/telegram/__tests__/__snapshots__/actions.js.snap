// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sendListingsToUser 1`] = `
Array [
  Array [
    "123124",
    Object {
      "source": Object {
        "data": Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        "type": "Buffer",
      },
    },
    Object {
      "caption": "https://av.by/kia/ceed/1231232",
    },
  ],
  Array [
    "123124",
    Object {
      "source": Object {
        "data": Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        "type": "Buffer",
      },
    },
    Object {
      "caption": "https://ab.onliner.by/cars/12314243214",
    },
  ],
]
`;
